From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Tue, 26 Jun 2012 19:28:14 +0200
Subject: Do not fail the /thread/thread4 test if prlimit() fails

This happens on the Debian buildds.

[smcv: Use g_test_skip()]

Forwarded: no, Debian buildd specific
---
 glib/tests/thread.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/glib/tests/thread.c b/glib/tests/thread.c
index b9f8796..4a9766c 100644
--- a/glib/tests/thread.c
+++ b/glib/tests/thread.c
@@ -148,7 +148,14 @@ test_thread4 (void)
   nl.rlim_cur = 1;
 
   if ((ret = prlimit (getpid (), RLIMIT_NPROC, &nl, &ol)) != 0)
-    g_error ("prlimit failed: %s", g_strerror (errno));
+    {
+      gchar *message = g_strdup_printf ("setting PRLIMIT_NPROC to {cur=%d,max=%d} failed: %s",
+                                        (int) nl.rlim_cur, (int) nl.rlim_max,
+                                        g_strerror (errno));
+      g_test_skip (message);
+      g_free (message);
+      return;
+    }
 
   error = NULL;
   thread = g_thread_try_new ("a", thread1_func, NULL, &error);
